posts/2023-01-20-types-announcement.md

```diff
@@ -0,0 +1,120 @@
+
+Despite the considerable effort put in over the years, we haven't *quite* achieved the key requirements listed above. To put some numbers to this, there are currently around 220 open tracking issues for [language](https://github.com/rust-lang/rust/issues?q=is%3Aopen+is%3Aissue+label%3AC-tracking-issue+label%3AT-lang), [compiler](https://github.com/rust-lang/rust/issues?page=1&q=is%3Aopen+is%3Aissue+label%3AC-tracking-issue+label%3AT-compiler), or [types](https://github.com/rust-lang/rust/issues?q=is%3Aopen+is%3Aissue+label%3AC-tracking-issue+label%3AT-types) features that have been accepted but are not completely implemented, of which about half are at least 3 years old and many are several years older than that. This is on top of the "deferred" RFCs that were never accepted: not because the idea itself wasn't appealing, but because of the lack of bandwidth to see the features through to stabilization.
+
+Another number that might shock you: there are currently 62 open [unsoundness issues](https://github.com/rust-lang/rust/issues?q=is%3Aopen+is%3Aissue+label%3AI-unsound). This sounds *much* scarier than it really is: nearly all of these are edges of the compiler and language that have been found by people who specifically poke and prod to find them; in practice these will not pop up in the programs you write. Nevertheless, these are edges we want to iron out.
+
+## The Types Team
+
+Moving forward, let's talk about a smaller subset of Rust rather than the entire language and compiler. Specifically, the parts relevant here include the type checker - loosely, defining the semantics and implementation of how variables are assigned their type, trait solving - deciding what traits are defined for which types, and borrow checking - proving that Rust's ownership model always holds. All of these can be thought of cohesively as the "type system".
+
+As of [RFC 3254](https://rust-lang.github.io/rfcs/3254-types-team.html), the above subset of the Rust language and compiler are under the purview of the types team. So, what exactly does this entail?
+
+First, since around 2018, there existed the "traits working group", which had the primary goal of creating a performant and extensible definition and implemention of Rust's trait system (including the [Chalk](https://github.com/rust-lang/chalk) trait-solving library). As time progressed, and particularly in the latter half of 2021 into 2022, the working group's influence and responsibility naturally expanded to the type checker and borrow checker too - they are actually strongly linked and its often hard to disentangle the trait solver from the other two. So, in some ways, the types team essentially subsumes the former traits working group.
```

Comment:
```suggestion
First, since around 2018, there existed the "traits working group", which had the primary goal of creating a performant and extensible definition and implementation of Rust's trait system (including the [Chalk](https://github.com/rust-lang/chalk) trait-solving library). As time progressed, and particularly in the latter half of 2021 into 2022, the working group's influence and responsibility naturally expanded to the type checker and borrow checker too - they are actually strongly linked and its often hard to disentangle the trait solver from the other two. So, in some ways, the types team essentially subsumes the former traits working group.
```